### DISCUSSION TITLE
Faster way to label faces

### DISCUSSION THREAD

**Discussion by nep2ner** [immich-app/immich#2558] [General]
I know that the face detection is already a very new feature, but I think that there could be a faster way to label faces. The current process lags.

Here is the current flow on the explore page: "view all" -> scroll down to find a face that isn't yet labeled (this page uses a lot of system resources when loading) -> click into that face -> label face -> click the back button which jumps you back to the explore page home -> click "view all" again to find the next face to label (using a bunch of system resources to load the page again).

I think that if the back button took the user back to the "view all" page instead of the explore home, it would make things smoother. Also, the "view all" page could be cached during that time to increase the load speed.

Another idea is to create a totally new UI flow that is specifically designed for bulk face labeling. It could show one face at a time and have the text input pre focused for user input, advancing to the next face on "enter" key press. If no name is typed before enter is pressed, it just skips and gives you a new face.

**Comment by tsaoi** [comment]
Great suggestion and I wonder if any progress has been made in this space. I'm a new user myself and am trying to determine the most efficient method to bulk tag faces without the back & forth screens.

**Comment by Melaron** [comment]
@tsaoi I am also new to Immich and am interested in a faster way to do this. What @nep2ner suggested a long time ago sounds good. I would love to click "Bulk assign names" and have Immich quickly let me type a name, press enter, and provide me a new face to name.

That said, it seems things have improved a bit since May 2023 and you can name a little faster (in web version). Go to Explore >> View all (small print, right side). You will then see all faces and when you hover over a face, a "three-dot" or "kabob" menu will appear (it is hard to see sometimes) and you can select it and then select "Change name" even if there is no name. Type the name and hit enter and you are still on the same page, at the same spot, and can now do the same thing for the next faces.

This is not ideal, but much better than moving pages every time.
